python/llm/src/bigdl/llm/transformers/models/baichuan2.py

```diff
@@ -301,3 +298,247 @@ def baichuan_attention_forward_13b(
+        ]
+    return mask
+
+
+def baichuan_13b_model_forward(
+    self,
+    input_ids: torch.LongTensor = None,
+    attention_mask: Optional[torch.Tensor] = None,
+    past_key_values: Optional[List[torch.FloatTensor]] = None,
+    inputs_embeds: Optional[torch.FloatTensor] = None,
+    use_cache: Optional[bool] = False,
+    output_attentions: Optional[bool] = False,
+    output_hidden_states: Optional[bool] = False,
+    return_dict: Optional[bool] = True,
+) -> Union[Tuple, BaseModelOutputWithPast]:
+    if input_ids is not None and inputs_embeds is not None:
+        raise ValueError(
+            "You cannot provide both input_ids and inputs_embeds simultaneously"
+        )
+    elif input_ids is not None:
+        batch_size, seq_length = input_ids.shape
+    elif inputs_embeds is not None:
+        batch_size, seq_length, _ = inputs_embeds.shape
+    else:
+        raise ValueError("You need to provide input_ids or inputs_embeds")
+
+    if self.first_run:
+        self.max_cache_pos = seq_length
+
+    return_dict = (
+        return_dict if return_dict is not None else self.config.use_return_dict
+    )
+
+    seq_length_with_past = seq_length
+
+    if past_key_values is not None:
+        past_key_values_length = past_key_values[0][0].shape[2]
+        seq_length_with_past = seq_length_with_past + past_key_values_length
+
+    if inputs_embeds is None:
+        inputs_embeds = self.embed_tokens(input_ids)
+
+    if self.training:
+        if (
+            self.alibi_mask is None
+            or self.alibi_mask.shape[-1] != seq_length_with_past
+        ):
+            self.alibi_mask = self.get_alibi_mask(
+                inputs_embeds, seq_length_with_past
+            )
+        alibi_mask = self.alibi_mask
+    else:
+        alibi_mask = self.get_alibi_mask(inputs_embeds, seq_length_with_past)
+
+    if attention_mask is not None:
+        if len(attention_mask.shape) == 2:
+            expanded_mask = attention_mask.to(alibi_mask.dtype)
+            expanded_mask = torch.tril(
+                torch.gt(expanded_mask[:, :, None] * expanded_mask[:, None, :], 0)
+            ) * torch.eq(expanded_mask[:, :, None] - expanded_mask[:, None, :], 0)
+        else:
+            expanded_mask = attention_mask
+        bsz = inputs_embeds.size(0)
+        src_len, tgt_len = alibi_mask.size()[-2:]
+        expanded_mask = (
+            expanded_mask.unsqueeze(1)
+            .expand(bsz, 1, src_len, tgt_len)
+            .to(alibi_mask.dtype)
+        )
+        inverted_mask = 1.0 - expanded_mask
+        inverted_mask = inverted_mask.masked_fill(
+            inverted_mask.to(torch.bool), torch.finfo(alibi_mask.dtype).min
+        )
+        attention_mask = inverted_mask + alibi_mask.unsqueeze(0)
+        if input_ids.device.type == "xpu":
+            del alibi_mask
+            del inverted_mask
+            del expanded_mask
+            torch.xpu.empty_cache()
```

Comment:
These mask variables are no longer used and deleting them can save some memory. Does it make sense? @jason-dai